src/shared/debug/dbgutil/dbgutil.cpp

```diff
@@ -271,18 +271,14 @@ HRESULT GetNextLevelResourceEntryRVA(ICorDebugDataTarget* pDataTarget,
     IMAGE_RESOURCE_DIRECTORY resourceDirectory;
     hr = ReadFromDataTarget(pDataTarget, moduleBaseAddress + resourceDirectoryRVA, (BYTE*)&resourceDirectory, sizeof(resourceDirectory));
 
-
-
-    // The ith resource directory entry is at offset 16 + 8i from the beginning of the resource
-    // directory table
-    WORD numNameEntries;
-    WORD numIDEntries;
     if (SUCCEEDED(hr))
     {
-        numNameEntries = resourceDirectory.NumberOfNamedEntries;
-        numIDEntries = resourceDirectory.NumberOfIdEntries;
+        // The ith resource directory entry is at offset 16 + 8i from the beginning of the resource
+        // directory table. First come named entries, followed by ID entries. We skip the former.
+        WORD idEntriesFirstIndex = resourceDirectory.NumberOfNamedEntries;
+        WORD idEntriesFinalIndex = (WORD)(idEntriesFirstIndex + resourceDirectory.NumberOfIdEntries);
```

Comment:
Calling this finalIndex sounds a bit misleading to me. The name sounds like it is inclusive, ie we'd iterate all indicies in the closed interval [firstIndex, finalIndex]. It may make the reader mistakenly think the for loop has an off-by-one error in it. If the goal of the rename is to increase clarity I'd suggest:

```
WORD countIDEntries = resourceDirectory.NumberOfIdEntries;
for(WORD i = idEntriesFirstIndex; i < idEntriesFirstIndex + countIDEntries; i++)
{
```

Reply:
The important part is the explicit cast. It's telling the compiler we know what we are doing with potential overflow of the sum. I can rename to beginIdEntries and endIdEntries if it's clearer to have the classic cpp begin and end pattern.

Reply:
begin and end would be fine, or include the explicit cast in the for loop conditional:
```
for(WORD i = idEntriesFirstIndex; i < (WORD)(idEntriesFirstIndex + countIDEntries); i++)
```